### PR TITLE
Created text and button components

### PR DESCRIPTION
**Pull request by thumn**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

# What's new in this PR
Created text and button components, largely copied with minor modifications from the customer app.

[//]: # "Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

## Relevant Links
N/A

### Online sources
- [Documentation on styled components](https://styled-components.com/docs/basics#adapting-based-on-props)

[//]: # "Optional - copy links to any tutorial or documentation that was useful to you when working on this PR"

### Related PRs
- [New reusable base components (in customer app)](https://github.com/calblueprint/dccentralkitchen/pull/28)

[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

## How to review
Review `BaseComponents.js` for component code.

Test by implementing base components in the app. Behavior should be exactly the same as components on the customer side.

[//]: # "The order in which to review files and what to expect when testing locally"

## Next steps
- Checkout button is not functional and should be implemented in the checkout screen.

[//]: # "What's NOT in this PR, doesn't work yet, and/or still needs to be done"

## Tests Performed, Edge Cases
Imported button container and button label into clerk login screen. Button should match the ones provided on Figma under Clerk components.

[//]: # "Hopefully we will add a testing suite/CI soon, but until then note down the steps you took to test locally"

### Screenshots
IMG_2363

[//]: # "Add screenshots of expected behavior - GIFs if you're feeling fancy!"

CC: @anniero98 @wangannie

[//]: # "This tags in both Annies as a default. Feel free to change, or add on anyone who you should be in on the conversation."

### CODE REVIEW

File: components/BaseComponents.js [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import styled from 'styled-components/native';
+import Colors from '../assets/Colors';
+
+// Text Components
+export const Title = styled.Text`
+  font-family: poppins-medium;
+  font-size: 20px;
+  line-height: 30px;
+  color: ${props => (props.color ? props.color : Colors.black)};
```
Comment (wangannie) [inline components/BaseComponents.js]: Can we change this everywhere to `${props => props.color || Colors.black}`? It's just a cleaner way to do this suggested by @emilyzhong

File: components/BaseComponents.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  text-transform: uppercase;
+  color: ${props => (props.color ? props.color : Colors.black)};
+`;
+
+export const TabSelected = styled.Text`
+  font-family: poppins-semibold;
+  font-size: 14px;
+  line-height: 20px;
+  color: ${props => (props.color ? props.color : Colors.black)};
+`;
```
Comment (wangannie) [inline components/BaseComponents.js]: Pretty sure we don't need this on Clerk since we don't use that style of tabs like on the customer app.

File: components/BaseComponents.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+  align-items: center;
+  text-align: center;
+  letter-spacing: 0.01;
+  text-transform: uppercase;
+  color: ${Colors.lightest};
+`;
+
+export const FilledButtonContainer = styled(ButtonContainer)`
+  width: 309px;
+  height: 57px;
```
Comment (wangannie) [inline components/BaseComponents.js]: Can we make width and height also take in props with a default value if none is specified like how it's done in Customer? 

`width: ${props => (props.width ? props.width : '309px')};
` and `height: ${props => (props.height ? props.height : '57px')};`

Comment (wangannie) [reply components/BaseComponents.js]: But also in the `${props => props.width || '309px'}` format

File: components/BaseComponents.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+export const ButtonLabel = styled.Text`
+  top: 25%;
+  bottom: 25%;
+  font-family: poppins-semibold;
+  font-size: 14px;
+  line-height: 21px;
+  display: flex;
+  align-items: center;
+  text-align: center;
+  letter-spacing: 0.01;
```
Comment (anniero98) [inline components/BaseComponents.js]: Nit: I think Expo will throw a warning if you don't give units ;o As in `letter-spacing: 0.01px` (if it's px?)

Comment (thumn) [reply components/BaseComponents.js]: ah good catch thanks!

File: lib/checkoutUtils.js [hunk 5]
```diff
@@ -13,27 +13,15 @@ export const categories = [
...
-      cartCount: 0
-    };
-  }
+  return {
+    name: record['Name'],
+    id: record.id,
+    category: record['Category'],
+    points: record['Points'],
+    customerCost: record['Customer Cost'],
+    image: record['Image'] ? record['Image'][0].url : null,
```
Comment (anniero98) [inline lib/checkoutUtils.js]: I modified this in my PR (since i swapped out all the Airtable calls) but doesn't conflict -- will adapt this in my branch (-:
